lms/static/js/discovery/app.js

```diff
@@ -3,21 +3,32 @@
 define(['backbone'], function(Backbone) {
     'use strict';
 
-    return function (Collection, Form, ResultListView, searchQuery) {
+    return function (Collection, Form, ResultListView, FilterBarView, FacetsBarView, searchQuery) {
 
         var collection = new Collection([]);
         var results = new ResultListView({ collection: collection });
         var dispatcher = _.clone(Backbone.Events);
         var form = new Form();
+        var filters = new FilterBarView();
+        var facets = new FacetsBarView();
 
         dispatcher.listenTo(form, 'search', function (query) {
-            collection.performSearch(query);
+            form.showLoadingIndicator();
+            filters.changeQueryFilter(query);
+        });
+
+        dispatcher.listenTo(filters, 'search', function (searchTerm, filters) {
+            collection.performSearch(searchTerm, filters);
             form.showLoadingIndicator();
         });
 
         dispatcher.listenTo(form, 'clear', function () {
             results.clearResults();
-            form.hideClearAllButton();
+            filters.hideClearAllButton();
+        });
+
+        dispatcher.listenTo(filters, 'clear', function () {
+            form.clearAll();
```

Comment:
Maybe just have this one handler and call `form.clearSearch` and `filters.hideClearAllButton` here?